### DISCUSSION TITLE
Athena Caching

### DISCUSSION THREAD

**Discussion by alwem82** [aws/aws-sdk-pandas#1248] [Q&A]
Hello,

like wrangler api for athena caching like below, 
is there similar way to execute the query in athena console with caching function?

wrangler api is the only way to enable athena caching?

<wrangler api for athena caching>
wr.athena.read_sql_query(query, database="awswrangler_test", max_cache_seconds=900, max_cache_query_inspections=500)

**Comment by jaidisido** [comment]
The AWS Athena service caches all query executions. Wrangler simply uses a combination of Athena API calls (ListQueryExecutions and BatchGetQueryExecutions) to simplify the process. AFAIK, the service does not offer a similar functionality however.

**Comment by 3atshan** [comment]
Hi on that note, is there a way to clear the Athena cache using wr or other library?

**Reply by malachi-constant** [reply]
There is no way to explicitly clear cache via the AWS SDK for pandas or boto3. The only potential workaround would be to clear the [S3 bucket](https://docs.aws.amazon.com/athena/latest/ug/querying.html) used to store athena results.
